src/cmd/build.js

```diff
@@ -55,9 +55,21 @@ async function defaultPackageCreator(
   let buffer = await zipDir(sourceDir, {
     filter: (...args) => fileFilter.wantFile(...args),
   });
-
+  let messageData: any;
+  if (manifestData.default_locale) {
+    messageData = readFileSync(
+     path.join(sourceDir, '_locales', manifestData.default_locale,
+     'messages.json'));
+  }
+  let extensionName: string;
+  if (messageData) {
+    extensionName = JSON.parse(messageData).extensionName.message;
```

Comment:
Since you need to add a new test case, it will help to break this block of code into a separate "unit" so it can be tested in isolation more easily. To do that, move all of this code to a new `async` function and call it like this:

````javascript
let extensionName: string;
if (manifestData.default_locale) {
  extensionName = await getDefaultLocalizedName({manifestData, sourceDir});
} else {
  extensionName = manifestData.name;
}
````
This will allow you to test for JSON syntax errors without calling the entire `build()` function.